pkg/router/template/router.go

```diff
@@ -497,16 +466,11 @@ func (r *templateRouter) writeConfig() error {
 
 	log.V(4).Info("router certificate manager config committed")
 
-	pathNames := make([]string, 0)
-	for k := range r.templates {
-		pathNames = append(pathNames, k)
-	}
-	sort.Strings(pathNames)
```

Comment:
I'm trying to understand why this was added in the first place.  It looks like https://github.com/openshift/origin/pull/19219/commits/1cfd4fde6c6a48c7df9d52fe034cf15424375382 added logic that wrote temporary map files under `/var/lib/haproxy/conf/.tmp/` and then used them to write the final map files under `/var/lib/haproxy/conf/`, so the sort was needed to make sure the `.tmp/` files were written first.  However, https://github.com/openshift/origin/pull/19492/commits/9e88ee03fca132c2aa0f5713031c4cf474d8cc21 removed the temporary files, at which point the sorting was no longer needed.  So this looks safe to remove.

Reply:
Ha, interesting. Yeah, just seeing the code as it stood made no sense and so the refactoring seemed harmless. Thanks for contextualizing it.